### PR TITLE
VACMS-9203: Introduce loadtesting framework for authenticated requests with Goose

### PR DESCRIPTION
**Pull request by ElijahLynn**
## Description
See https://github.com/department-of-veterans-affairs/va.gov-cms/issues/9203

## QA steps

What needs to be checked to prove this works?  
What needs to be checked to prove it didn't break any related things?  
What variations of circumstances (users, actions, values) need to be checked?  

As user _uid_ with _user_role_
1. Do this
   - [ ] Validate that
2. Then
   - [ ] Validate that
3. Then validate Acceptance Criteria from issue
   - [ ] a
   - [ ] b
   - [ ] c

### Definition of Done

- [ ] Documentation has been updated, if applicable.
- [ ] Tests have been added if necessary.
- [ ] Automated tests have passed.
- [ ] Code Quality Tests have passed.
- [ ] Acceptance Criteria in related issue are met.
- [ ] Manual Code Review Approved.

### Select Team for PR review

- [ ] `Platform CMS Team`
- [ ] `Sitewide program`
- [ ] `⭐️ Sitewide CMS`
- [ ] `⭐️ Public websites`
- [ ] `⭐️ Facilities`
- [ ] `⭐️ User support`

### Is this PR blocked by another PR?

- [ ] `DO NOT MERGE`

### Does this PR need review from a Product Owner

- [ ] `Needs PO review`

### CMS user-facing annoucement

Is an announcement needed to let editors know of this change?
- [ ] Yes, and it's written in issue ____ and queued for publication.
  - [ ] Merge and ping @ rachel-kauff so she's ready to publish after deployment
- [ ] Yes, but it hasn't yet been written
  - [ ] Don't merge yet -- ping @ rachel-kauff to prompt her to write and queue content
- [ ] No announcement is needed for this code change.
  - [ ] Merge & carry on unburdened by announcements

### CODE REVIEW

File: tests/loadtest/src/admin.rs [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+pub async fn log_in(user: &mut GooseUser) -> TransactionResult {
+    // Use ADMIN_USERNAME= to set custom admin username.
+    let admin_username = match env::var("ADMIN_USERNAME") {
+        Ok(username) => username,
+        Err(_) => "admin".to_string(),
+    };
+    // Use ADMIN_PASSWORD= to set custom admin username.
+    let admin_password = match env::var("ADMIN_PASSWORD") {
+        Ok(password) => password,
+        Err(_) => "P@ssw0rd1234".to_string(),
```
Comment (cweagans) [inline tests/loadtest/src/admin.rs:15]: This seems like something we shouldn't include. Even if it's not a real password, probably better to just throw an error if the env vars aren't set.

File: tests/loadtest/src/main.rs [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        // .register_scenario(
+        //     scenario!("Anonymous English user")
+        //         .set_weight(40)?
+        //         // .set_wait_time(Duration::from_secs(0), Duration::from_secs(1))?
+        //         .register_transaction(
+        //             transaction!(benefits_detail_page)
+        //                 .set_name("anon /")
+        //                 .set_weight(2)?,
+        //         ),
+        // )
```
Comment (cweagans) [inline tests/loadtest/src/main.rs:25]: Can you please remove the commented code?

File: tests/loadtest/src/main.rs [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+                    transaction!(log_in)
+                        .set_on_start()
+                        .set_name("auth login"),
+                )
+                .register_transaction(
+                    transaction!(benefits_detail_page)
+                      .set_name("auth /disability/how-to-file-claim"),
+                ),
+        )
+        .set_default(GooseDefault::Host, "https://drupal-9.ddev.site/")?
```
Comment (cweagans) [inline tests/loadtest/src/main.rs:40]: Not sure this should have a default. I think it'd be okay to throw an error if this isn't set by the user? The CMS doesn't run at `drupal-9.ddev.site` normally, so this won't work for anyone at our team, right?

**Review by cweagans — DISMISSED** [review]
Approved with a couple of quick (?) changes. If they aren't quick, don't worry about them.

File: tests/loadtest/src/main.rs (hunk 2 above)
Comment (ElijahLynn) [reply tests/loadtest/src/main.rs:25]: Yeah, I do want to add it back to be able to just run anon vs auth scenario but haven't figured out how to do that yet. Will delete for now.

File: tests/loadtest/src/main.rs (hunk 3 above)
Comment (ElijahLynn) [reply tests/loadtest/src/main.rs:40]: Correct, won't work for anyone, it came from the default in the goose-eggs crate. I think we should change to va-gov-cms.ddev.site.

File: tests/loadtest/src/admin.rs (hunk 1 above)
Comment (ElijahLynn) [reply tests/loadtest/src/admin.rs]: Changed to be the export commands needed. Not sure how else to do it right now, the Err() needs to be there (tested).

**Pull request merged**